head.S

```diff
@@ -183,6 +183,9 @@ GLOBAL(_entry)
 	cmpl	$MULTIBOOT2, boot_protocol(%ebp)
 	je	multiboot2_kernel
 
+	cmpl	$SIMPLE_PAYLOAD, boot_protocol(%ebp)
+	je	simple_payload
+
```

Comment:
While the logic here is fine, the asm is suboptimal.  It would be better as:

```asm
    mov    boot_protocol(%ebp), %edx
    cmp    $MULTIBOOT2, %edx
    je     multiboot2_kernel

    cmp    $SIMPLE_PAYLOAD, %edx
    je     simple_payload
```

Reply:
Changed. Saves us 2 bytes :slightly_smiling_face:

Reply:
And an extra memory read.